pkg/apis/apps/v1alpha1/apimanager_types.go

```diff
@@ -519,6 +528,7 @@ func (apimanager *APIManager) setSystemFileStorageSpecDefaults() (bool, error) {
 
 	if systemSpec.FileStorageSpec != nil &&
 		systemSpec.FileStorageSpec.PVC != nil &&
+		systemSpec.FileStorageSpec.DeprecatedS3 != nil &&
```

Comment:
DeprecatedS3 should never be used but during the migration it is set so we manage that case and make sure that only one of them is used

Reply:
`setSystemFileStorageSpecDefaults` is part of the reconciliation and you can assume it has been migrated. No need to check

Reply:
Even though it's been migrated there's nothing preventing that it is set by the user (or by the operator itself by mistake) after the migration. You do the same between checking PVC and S3. You just make sure that one of the fields in that section is set (a union).

Reply:
As I see, in reconciliation loop the operator does not care anymore on that field.

I would add a check to see that DeprecateS3 is not null and when that is the case, report some warning using logs, nothing else.

Reply:
Added a log line in system_reconciler.go file. Please review it.